pythonforandroid/python.py

```diff
@@ -291,9 +296,8 @@ def create_python_bundle(self, dirn, arch):
         python_lib_name = 'libpython' + self.major_minor_version_string
         if self.major_minor_version_string[0] == '3':
             python_lib_name += 'm'
-        for lib in [python_lib_name + '.so', python_lib_name + '.so.1.0']:
-            shprint(sh.cp, join(python_build_dir, lib),
-                    'libs/{}'.format(arch.arch))
+        shprint(sh.cp, join(python_build_dir, python_lib_name + '.so'),
```

Comment:
I don't understand, why would you remove the loop here? The only thing that seems to achieve is that in case it is ever named `.so.1.0` again in the future (which I assume can happen depending on NDK or compiler version) it will break. Or does this actually fix something?

Reply:
If we don't remove the versioned library we will get an error when running the copy command, because now the versioned library doesn't exist anymore

Reply:
@opacam I'm just thinking, is this make option guaranteed to exist in the future? Is it some standard thing? Maybe it would be better to make the `for lib` loop based on `os.listdir` instead?

Reply:
Also is this fixing something broken by this pull request? If not, better make a separate one. This one is huge enough and the more things are added on top, the longer it will take us to look through it and get it merged :grimacing:

Reply:
> I'm just thinking, is this make option guaranteed to exist in the future?

Nothing is guaranteed to exist in the future  :laughing: :laughing:...but this option exists at least since the python version 2.7.2...we have it in our python2legacy recipe...so...probably will continue there for a while...

Reply:
> Also is this fixing something broken by this pull request?...

No, but it will cause troubles, and we know it...so I though that it will be better get fixed as soon as possible because it requires only a few lines, plus the affected files are already touched for this pr, it affects directly the work done in here and moreover...there still some changes that must be reviewed...**but you are also right...better to  not be touched anymore unless requested.**

Reply:
That is a non-issue, we have code comments for that :wink: (you could think that with the old loop as well)

I don't think it would be much longer than the previous code, and avoid that Makefile option. I guess using the Makefile option is kind of neater, but I'm not sure if we should trust it to stay around...? Might be worth checking in the Makefile or python install docs if this is something official or rather some internal variable we hacked into that isn't meant to be used from the outside

**Anyway**, in conclusion I would suggest you remove this commit, and make a separate pull request (since this is only bringing up new discussion and isn't strictly part of this, we should stop delaying things)

Reply:
> but you are also right...better to not be touched anymore unless requested.

true, and I would actually suggest you undo that latest commit and make a separate pull request. it's no good to stuff more things into this, even if they are necessary

> will be better get fixed as soon as possible

Even more of a reason to make a separate pull request! piling up more things into a single one we won't get it in any sooner due to the complexity of all the changes

Reply:
**INSTNOME It's official**, and it has been there for a long, long, long time...checking the cpython code I see that was introduced at year 2002 (python/cpython@1142de3)...seems to me that has been pretty well tested, even for us because @tito make use of it in his original recipe (bdefea1) and @inclement did the same with his rework of the recipe (when the recipe.sh was removed, 4c8b5bc) so...we are currently having this discussion thanks to this variable jajajaja...at the point that we remove it in python3 recipe... we begin to have this linkage problems...so I have no doubt that we can rely on this INSTNOME variable...don't you think?

About removing the commit...mmmm...I already shot the bullet...even if I remove the commit...the damage has been done...lets the administrator team decide ok  :wink:?

**Note:** If we create the pr you mention and gets merged before this, we may have troubles resolving the merge conflicts in here and we could delay this even more...that is the reason to not  create the pr in first place...and the reason to create a proper patch and post it into the proper issue and to notify it in discord.

Reply:
> About removing the commit...mmmm...I already shot the bullet...even if I remove the commit...the damage has been done...lets the administrator team decide ok wink?

not really, the commits will all be squashed anyway, so if you do a revert it will be truly removed with no issues. there is really no problem if you just revert it.

you could already squash the commit history as a preparation if you want, it would need to be done on the final merge with github's squash button anyway